src/modules/logger/logger.cpp

```diff
@@ -515,51 +508,32 @@ void Logger::run()
 
 						//PX4_INFO("topic: %s, size = %zu, out_size = %zu", sub.metadata->o_name, sub.metadata->o_size, msg_size);
 
-#ifdef DBGPRINT
-						//warnx("subscription %s updated: %d, size: %d", sub.metadata->o_name, updated, msg_size);
-						hrt_abstime trytime = hrt_absolute_time();
-
-						if (_writer._count > highWater) {
-							highWater = _writer._count;
-						}
-
-#endif /* DBGPRINT */
-
 						if (_writer.write(buffer, msg_size)) {
 
 #ifdef DBGPRINT
+							total_bytes += msg_size;
+#endif /* DBGPRINT */
 
-							// successful write: note end of dropout if dropout_start != 0
-							if (dropout_start != 0) {
-								double drop_len = (double)(trytime - dropout_start)  * 1e-6;
+							if (_dropout_start) {
+								float dropout_duration = (float)(hrt_elapsed_time(&_dropout_start) / 1000) / 1.e3f;
 
-								if (drop_len > max_drop_len) {
-									max_drop_len = drop_len;
+								if (dropout_duration > _max_dropout_duration) {
+									_max_dropout_duration = dropout_duration;
 								}
 
-								PX4_WARN("dropout length: %5.3f seconds", drop_len);
-								dropout_start = 0;
-								highWater = 0;
+								_dropout_start = 0;
 							}
 
-							total_bytes += msg_size;
-#endif /* DBGPRINT */
-
 							data_written = true;
 
 						} else {
 
-#ifdef DBGPRINT
-
-							if (dropout_start == 0)	{
-								available = _writer._count;
-								PX4_WARN("dropout, available: %zu/%zu", available, _writer._buffer_size);
-								dropout_start = trytime;
-								dropout_count++;
+							if (!_dropout_start) {
+								_dropout_start = hrt_absolute_time();
+								++_write_dropouts;
+								_high_water = 0;
```

Comment:
Once we get a dropout, highwater becomes meaningless (it would be >buffer size), so I reset it. What we could do is to keep track by how much the buffer was overflown.